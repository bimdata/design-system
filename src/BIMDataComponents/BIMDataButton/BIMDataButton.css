.bimdata-btn {
  height: fit-content;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: transparent;
  border: none;
  color: var(--color-primary);
  cursor: pointer;
  font-family: var(--primary-font);
  font-size: 12px;
  transition: all 0.1s ease;
  &:hover {
    transition: all 0.25s ease-in;
  }
  &:focus {
    outline: none;
  }
  /* style BIMDATA FONT-WEIGHT */
  &.light {
    font-weight: 300;
  }
  &.bold {
    font-weight: 700;
  }
  /* style BIMDATA BTN RADIUS ------------------ */
  &.radius {
    border-radius: 6px;
  }
  &.square {
    border-radius: 0;
  }
  &.rounded {
    border-radius: 50px;
  }
  /* style BIMDATA BTN FILL -------------------- */
  &.fill {
    background-color: var(--color-white);
    justify-content: center;
    /* style BIMDATA BTN DEFAULT FILL ---------- */
    &.default {
      background-color: #e9eff8;
      color: var(--color-primary);
      /* &:hover {
        background-color: var(--color-silver-light);
      } */
    }
    /* style BIMDATA BTN LIGHT-DEFAULT FILL ---------- */
    &.light-default {
      background-color: #E9EFF8;
      color: var(--color-tertiary);
    }
    /* style BIMDATA BTN PRIMARY FILL ---------- */
    &.primary {
      background-color: var(--color-primary);
      color: var(--color-white);
      &:hover {
        background-color: var(--color-primary-light);
      }
    }
    /* style BIMDATA BTN SECONDARY FILL ---------- */
    &.secondary {
      background-color: var(--color-secondary);
      color: var(--color-primary);
      &:hover {
        background-color: var(--color-secondary-light);
      }
    }
    /* style BIMDATA BTN TERTIARY FILL ---------- */
    &.tertiary {
      background-color: var(--color-tertiary);
      color: var(--color-white);
      /* &:hover {
        background-color: var(--color-tertiary-light);
      } */
    }
    /* style BIMDATA BTN GREY FILL ------------- */
    &.granite {
      background-color: var(--color-granite);
      color: var(--color-white);
      &:hover {
        background-color: var(--color-granite);
      }
    }
    /* style BIMDATA BTN RED FILL ------------- */
    &.high {
      background-color: var(--color-high);
      color: var(--color-white);
      &:hover {
        background-color: var(--color-high-light);
      }
    }
    /* style BIMDATA BTN WARNING FILL ------------- */
    &.warning {
      background-color: var(--color-warning);
      color: var(--color-white);
      &:hover {
        background-color: var(--color-warning-light);
      }
    }
    /* style BIMDATA BTN GREEN FILL ------------- */
    &.success {
      background-color: var(--color-success);
      color: var(--color-white);
      &:hover {
        background-color: var(--color-success-light);
      }
    }
    /* style BIMDATA BTN NEUTRAL FILL ------------- */
    &.info {
      background-color: var(--color-neutral);
      color: var(--color-white);
      &:hover {
        background-color: var(--color-neutral-light);
      }
    }
    /* style BIMDATA BTN FILL BTN ICON --------- */
    &--btn-icon {
      svg {
        margin: 0 6px;
      }
    }
    &:disabled {
      background-color: var(--color-silver-dark);
      color: var(--color-white);
      cursor: auto;
    }
  }
  /* style BIMDATA BTN RIPPLE ----------------- */
  &.ripple {
    background-position: center;
    transition: background 0.8s;
    /* style BIMDATA BTN DEFAULT RIPPLE ------- */
    &.default {
      background-color: transparent;
      color: var(--color-primary);
      &:hover {
        background: transparent
          radial-gradient(circle, transparent 1%, var(--color-silver-light) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-silver-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    /* style BIMDATA BTN LIGHT DEFAULT RIPPLE ------- */
    &.light-default {
      background-color: transparent;
      color: var(--color-primary);
      &:hover {
        background: transparent
          radial-gradient(circle, transparent 1%, var(--color-silver-light) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-silver-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    /* style BIMDATA BTN PRIMARY RIPPLE ------- */
    &.primary {
      background-color: var(--color-primary);
      color: var(--color-white);
      &:hover {
        background: var(--color-primary)
          radial-gradient(circle, transparent 1%, var(--color-primary) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-primary-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &.secondary {
      background-color: var(--color-secondary);
      color: var(--color-primary);
      &:hover {
        background: var(--color-secondary)
          radial-gradient(circle, transparent 1%, var(--color-secondary) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-secondary-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &.tertiary {
      background-color: var(--color-tertiary);
      color: var(--color-white);
      &:hover {
        background: var(--color-tertiary)
          radial-gradient(circle, transparent 1%, var(--color-tertiary) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-tertiary-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &.granite {
      background-color: var(--color-granite);
      color: var(--color-white);
      &:hover {
        background: var(--color-granite)
          radial-gradient(circle, transparent 1%, var(--color-granite) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-granite-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &.high {
      background-color: var(--color-high);
      color: var(--color-white);
      &:hover {
        background: var(--color-high)
          radial-gradient(circle, transparent 1%, var(--color-high) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-high-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &.success {
      background-color: var(--color-success);
      color: var(--color-white);
      &:hover {
        background: var(--color-success)
          radial-gradient(circle, transparent 1%, var(--color-success) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-success-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &.warning {
      background-color: var(--color-warning);
      color: var(--color-white);
      &:hover {
        background: var(--color-warning)
          radial-gradient(circle, transparent 1%, var(--color-warning) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-warning-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &.info {
      background-color: var(--color-neutral);
      color: var(--color-white);
      &:hover {
        background: var(--color-neutral)
          radial-gradient(circle, transparent 1%, var(--color-neutral) 1%)
          center/15000%;
      }
      &:active:not(:disabled) {
        background-color: var(--color-neutral-light);
        background-size: 100%;
        transition: background 0s;
      }
    }
    &:disabled {
      background-color: var(--color-silver-dark);
      color: var(--color-white);
      cursor: auto;
    }
  }
  /* style BIMDATA BTN OUTLINE ----------------- */
  &.outline {
    justify-content: center;
    border: 1px solid;
    /* style BIMDATA BTN DEFAULT OUTLINE ------- */
    &.default {
      border-color: var(--color-tertiary-lighter);
      color: var(--color-primary);
      /* &:hover {
        background-color: var(--color-silver-light);
      } */
    }
    &.light-default {
      border-color: var(--color-tertiary-lighter);
      color: var(--color-tertiary);
      /* &:hover {
        background-color: var(--color-silver-light);
      } */
    }
    &.primary {
      border-color: var(--color-primary);
      color: var(--color-primary);
      &:hover {
        background-color: var(--color-primary-lighter);
      }
    }
    &.secondary {
      border-color: var(--color-secondary);
      color: var(--color-primary);
      &:hover {
        background-color: var(--color-secondary-lighter);
      }
    }
    &.tertiary {
      border-color: var(--color-tertiary);
      color: var(--color-tertiary);
      &:hover {
        background-color: var(--color-tertiary-lighter);
      }
    }
    &.granite {
      border-color: var(--color-granite);
      color: var(--color-granite);
      &:hover {
        background-color: var(--color-silver-light);
      }
    }
    &.high {
      border-color: var(--color-high);
      color: var(--color-high);
      &:hover {
        background-color: var(--color-high-lighter);
      }
    }
    &.success {
      border-color: var(--color-success);
      color: var(--color-success);
      &:hover {
        background-color: var(--color-success-lighter);
      }
    }
    &.warning {
      border-color: var(--color-warning);
      color: var(--color-warning);
      &:hover {
        background-color: var(--color-warning-lighter);
      }
    }
    &.info {
      border-color: var(--color-neutral);
      color: var(--color-neutral);
      &:hover {
        background-color: var(--color-neutral-lighter);
      }
    }
    &:disabled {
      border-color: var(--color-silver-dark);
      color: var(--color-silver-dark);
      cursor: auto;
      &:hover {
        background-color: transparent;
      }
    }
  }
  &.ghost {
    justify-content: center;
    &.default {
      color: var(--color-primary);
      &:hover {
        background-color: var(--color-silver-light);
      }
    }
    &.light-default {
      color: var(--color-primary);
      &:hover {
        background-color: var(--color-silver-light);
      }
    }
    &.primary {
      color: var(--color-primary);
      &:hover {
        background-color: var(--color-primary-lighter);
      }
    }
    &.secondary {
      color: var(--color-secondary);
      &:hover {
        background-color: var(--color-secondary-lighter);
      }
    }
    &.tertiary {
      color: var(--color-tertiary);
      &:hover {
        background-color: var(--color-tertiary-lighter);
      }
    }
    &.granite {
      color: var(--color-granite);
      &:hover {
        background-color: var(--color-silver-light);
      }
    }
    &.high {
      color: var(--color-high);
      &:hover {
        background-color: var(--color-high-lighter);
      }
    }
    &.success {
      color: var(--color-success);
      &:hover {
        background-color: var(--color-success-lighter);
      }
    }
    &.warning {
      color: var(--color-warning);
      &:hover {
        background-color: var(--color-warning-lighter);
      }
    }
    &.info {
      color: var(--color-neutral);
      &:hover {
        background-color: var(--color-neutral-lighter);
      }
    }
    &:disabled {
      color: var(--color-silver-dark);
      cursor: auto;
    }
  }
}
