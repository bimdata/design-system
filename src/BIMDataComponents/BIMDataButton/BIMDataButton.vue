<template>
  <button
    class="bimdata-btn"
    :class="classes"
    :style="style"
    @click="$emit('click', $event)"
    :color="color"
    type="button"
  >
    <slot></slot>
  </button>
</template>

<script>
import colors from "../../assets/colors.js";

export default {
  name: "BIMDataButton",
  props: {
    width: {
      type: String,
      default: "32px",
    },
    height: {
      type: String,
      default: "32px",
    },
    fill: {
      type: Boolean,
      default: false,
    },
    outline: {
      type: Boolean,
      default: false,
    },
    ghost: {
      type: Boolean,
      default: false,
    },
    ripple: {
      type: Boolean,
      default: false,
    },
    radius: {
      type: Boolean,
      default: false,
    },
    square: {
      type: Boolean,
      default: false,
    },
    rounded: {
      type: Boolean,
      default: false,
    },
    icon: {
      type: Boolean,
      default: false,
    },
    color: {
      type: String,
      default: "default",
      validator: color => colors.includes(color),
    },
  },
  emits: ["click"],
  computed: {
    classes() {
      return {
        "bimdata-btn__icon": this.icon,
        "bimdata-btn__fill": this.fill,
        "bimdata-btn__outline": this.outline,
        "bimdata-btn__ghost": this.ghost,
        "bimdata-btn__ripple": this.ripple,
        "bimdata-btn__radius": this.radius,
        "bimdata-btn__square": this.square,
        "bimdata-btn__rounded": this.rounded,
        [`bimdata-btn__fill--${this.color}`]: this.fill && this.color,
        [`bimdata-btn__outline--${this.color}`]: this.outline && this.color,
        [`bimdata-btn__ghost--${this.color}`]: this.ghost && this.color,
        [`bimdata-btn__ripple--${this.color}`]: this.ripple && this.color,
      };
    },
    style() {
      return {
        "min-width": `${this.width}`,
        "min-height": `${this.height}`,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
@import "../../assets/scss/_BIMDataVariables.scss";
@import "../../assets/scss/utilities/_text.scss";
</style>

<style lang="scss" src="./_BIMDataButton.scss"></style>
