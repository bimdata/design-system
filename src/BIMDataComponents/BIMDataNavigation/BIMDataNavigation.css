.bimdata-navigation {
  display: inline-flex;
  width: 100%;
  height: 38px;
  justify-content: center;
  align-items: center;
  .bimdata-navigation__item {
    padding: 6px 2px;

    .bimdata-btn {
      padding: 0;
      justify-content: center;
    }

    &.active {
      .bimdata-btn {
        font-weight: 700;
      }
    }
  }

  .bimdata-navigation__left,
  .bimdata-navigation__right {
    padding: 6px 2px;
    font-size: 10px;
    cursor: default;
  }

  &.dark {
    
  }
}
