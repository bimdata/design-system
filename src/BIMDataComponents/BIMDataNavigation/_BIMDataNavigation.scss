@import "../../assets/scss/_BIMDataVariables.scss";
@import "../../assets/scss/utilities/_list.scss";

.bimdata-navigation {
  display: inline-flex;
  width: 100%;
  height: 38px;
  justify-content: center;
  align-items: center;
  background-color: var(--color-white);
  &__item {
    padding: 6px 2px;

    .bimdata-btn {
      padding: 0;
      justify-content: center;
    }

    &.active {
      .bimdata-btn {
        font-weight: bold;
      }
    }
  }

  &__left,
  &__right {
    padding: 6px 2px;
    font-size: 10px;
  }
}
