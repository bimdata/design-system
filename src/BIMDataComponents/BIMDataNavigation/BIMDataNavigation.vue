<template>
  <ul class="bimdata-list bimdata-navigation">
    <li class="bimdata-navigation__left m-r-6">
      <slot name="left"></slot>
    </li>
    <li class="bimdata-navigation__item" v-if="first">
      <BIMDataButton
        @click="$emit('first')"
        :disabled="firstDisabled"
        ghost
        :class="{ disabled: firstDisabled }"
        width="21px"
        height="21px"
      >
        <BIMDataIcon name="doubleChevron" size="xxxs" :rotate="180" />
      </BIMDataButton>
    </li>
    <li class="bimdata-navigation__item" v-if="previous">
      <BIMDataButton
        @click="$emit('previous')"
        :disabled="previousDisabled"
        ghost
        :class="{ disabled: previousDisabled }"
        width="21px"
        height="21px"
      >
        <BIMDataIcon name="chevron" size="xxxs" :rotate="180" />
      </BIMDataButton>
    </li>
    <li>
      <slot />
    </li>
    <li class="bimdata-navigation__item" v-if="next">
      <BIMDataButton
        @click="$emit('next')"
        :disabled="nextDisabled"
        ghost
        :class="{ disabled: nextDisabled }"
        width="21px"
        height="21px"
      >
        <BIMDataIcon name="chevron" size="xxxs" />
      </BIMDataButton>
    </li>
    <li class="bimdata-navigation__item" v-if="last">
      <BIMDataButton
        @click="$emit('last')"
        :disabled="lastDisabled"
        ghost
        :class="{ disabled: lastDisabled }"
        width="21px"
        height="21px"
      >
        <BIMDataIcon name="doubleChevron" size="xxxs" />
      </BIMDataButton>
    </li>
    <li class="bimdata-navigation__right">
      <slot name="right"></slot>
    </li>
  </ul>
</template>

<script>
import BIMDataIcon from "../../BIMDataComponents/BIMDataIcon/BIMDataIcon.vue";
import BIMDataButton from "../../BIMDataComponents/BIMDataButton/BIMDataButton.vue";

export default {
  components: {
    BIMDataIcon,
    BIMDataButton,
  },
  props: {
    first: {
      type: Boolean,
      default: true,
    },
    last: {
      type: Boolean,
      default: true,
    },
    previous: {
      type: Boolean,
      default: true,
    },
    next: {
      type: Boolean,
      default: true,
    },
    firstDisabled: {
      type: Boolean,
      default: false,
    },
    lastDisabled: {
      type: Boolean,
      default: false,
    },
    previousDisabled: {
      type: Boolean,
      default: false,
    },
    nextDisabled: {
      type: Boolean,
      default: false,
    },
  },
  emits: ["first", "previous", "next", "last"],
};
</script>

<style lang="scss" scoped>
@import "../../assets/scss/_BIMDataVariables.scss";
@import "../../assets/scss/utilities/_list.scss";
</style>

<style scoped lang="scss" src="./_BIMDataNavigation.scss"></style>
