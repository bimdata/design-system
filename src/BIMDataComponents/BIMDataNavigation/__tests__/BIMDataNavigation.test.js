import { shallowMount } from "@vue/test-utils";
import BIMDataNavigation from "../BIMDataNavigation.vue";
import BIMDataButton from "../../BIMDataButton/BIMDataButton.vue";

describe("BIMDataNavigation", () => {
  it("should render component and match snapshot", () => {
    const wrapper = shallowMount(BIMDataNavigation);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a navigation with a left slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataNavigation, {
      slots: {
        left: "<span>left slot</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a navigation with a right slot and match snapshot", () => {
    const wrapper = shallowMount(BIMDataNavigation, {
      slots: {
        right: "<span>right slot</span>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render li tag", () => {
    const wrapper = shallowMount(BIMDataNavigation);
    expect(wrapper.find(".bimdata-navigation__item").element).toBeTruthy();
  });

  it("should not render BIMDataButton", () => {
    const wrapper = shallowMount(BIMDataNavigation, {
      propsData: {
        first: false,
        last: false,
        previous: false,
        next: false,
      },
    });
    expect(wrapper.findComponent(BIMDataButton).exists()).toBe(false);
  });

  it("should render BIMDataButton disabled", () => {
    const wrapper = shallowMount(BIMDataNavigation, {
      propsData: {
        firstDisabled: true,
        previousDisabled: true,
        nextDisabled: true,
        lastDisabled: true,
      },
    });
    expect(wrapper.findComponent(BIMDataButton).classes("disabled")).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should emit first event", async () => {
    const wrapper = shallowMount(BIMDataNavigation);
    wrapper.vm.$emit("first");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted("first")).toBeTruthy();
  });

  it("should emit previous event", async () => {
    const wrapper = shallowMount(BIMDataNavigation);
    wrapper.vm.$emit("previous");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted("previous")).toBeTruthy();
  });

  it("should emit next event", async () => {
    const wrapper = shallowMount(BIMDataNavigation);
    wrapper.vm.$emit("next");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted("next")).toBeTruthy();
  });

  it("should emit last event", async () => {
    const wrapper = shallowMount(BIMDataNavigation);
    wrapper.vm.$emit("last");
    await wrapper.vm.$nextTick();
    expect(wrapper.emitted("last")).toBeTruthy();
  });
});
