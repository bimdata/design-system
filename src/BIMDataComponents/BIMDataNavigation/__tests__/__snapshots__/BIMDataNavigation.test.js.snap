// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BIMDataNavigation should render BIMDataButton disabled 1`] = `
"<ul class=\\"bimdata-list bimdata-navigation\\">
  <li class=\\"bimdata-navigation__left m-r-6\\"></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" disabled=\\"true\\" class=\\"disabled\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" disabled=\\"true\\" class=\\"disabled\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" disabled=\\"true\\" class=\\"disabled\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" disabled=\\"true\\" class=\\"disabled\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__right\\"></li>
</ul>"
`;

exports[`BIMDataNavigation should render a navigation with a left slot and match snapshot 1`] = `
"<ul class=\\"bimdata-list bimdata-navigation\\">
  <li class=\\"bimdata-navigation__left m-r-6\\"><span>left slot</span></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__right\\"></li>
</ul>"
`;

exports[`BIMDataNavigation should render a navigation with a right slot and match snapshot 1`] = `
"<ul class=\\"bimdata-list bimdata-navigation\\">
  <li class=\\"bimdata-navigation__left m-r-6\\"></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__right\\"><span>right slot</span></li>
</ul>"
`;

exports[`BIMDataNavigation should render component and match snapshot 1`] = `
"<ul class=\\"bimdata-list bimdata-navigation\\">
  <li class=\\"bimdata-navigation__left m-r-6\\"></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"180\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li></li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataiconchevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataiconchevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__item\\">
    <bimdatabutton-stub width=\\"21px\\" height=\\"21px\\" ghost=\\"true\\" color=\\"default\\" class=\\"\\">
      <bimdataicondoublechevron-stub color=\\"default\\" fillcolor=\\"currentColor\\" size=\\"xxxs\\" rotate=\\"0\\" margin=\\"0px\\"></bimdataicondoublechevron-stub>
    </bimdatabutton-stub>
  </li>
  <li class=\\"bimdata-navigation__right\\"></li>
</ul>"
`;
